.github/workflows/test.yml

```diff
@@ -20,28 +22,62 @@ jobs:
       - name: Determine changed packages
         id: changed-packages
         run: |
-          CHANGED=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} | grep '^packages/' | awk -F'/' '{print $2}' | sort | uniq | tr '\n' ' ')
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            # Fetch the base branch
+            git fetch origin ${{ github.base_ref }}
+            BASE_REF="origin/${{ github.base_ref }}"
+            HEAD_REF="${{ github.event.pull_request.head.sha }}"
+          else
+            BASE_REF="${{ github.event.before }}"
+            HEAD_REF="${{ github.sha }}"
+          fi
+          echo "$BASE_REF - $HEAD_REF"
+          CHANGED=$(git diff --name-only $BASE_REF...$HEAD_REF | grep '^packages/' | awk -F'/' '{print $2}' | sort | uniq | tr '\n' ' ')
           echo "Changed packages: $CHANGED"
-          echo "::set-output name=list::$CHANGED"
+          echo "{list}={$CHANGED}" >> $GITHUB_OUTPUT
+
+      - name: Prepare test user for player tests
+        if: contains(steps.changed-packages.outputs.list, 'player')
+        id: json
+        run: |
+          echo ${{ env.TEST_USER }} | base64 -d > test-user.json
+          json=$(cat test-user.json)
+          replacement=${{ secrets.PLAYER_REFRESH_TOKEN }}
+          fixed_json="$(echo ${json/\[TOKEN\]/$replacement})"
+          encoded_json=$(echo "$fixed_json" | base64 | tr -d '\n')
+          echo "encoded=\"$encoded_json\"" >> $GITHUB_OUTPUT
+        env:
+          TEST_USER: ${{ secrets.PLAYER_TEST_USER }}
+
+      - name: Run Player Tests
+        if: contains(steps.changed-packages.outputs.list, 'player')
+        run: |
+          pnpm --filter player run test || exit 1
+        env:
+          TEST_USER: ${{ steps.json.outputs.encoded }}
 
       # Test unchanged packages
       - name: Test unchanged packages
         run: |
           ALL_PKGS=$(ls packages | tr '\n' ' ')
-          CHANGED_PKGS="${{ steps.changed-packages.outputs.list }}"
           for PKG in $ALL_PKGS; do
-            if [[ " $CHANGED_PKGS " != *" $PKG "* ]]; then
+            if [[ " ${{ env.CHANGED_PKGS }} " != *" $PKG "* ]] && [[ "$PKG" != "player" ]]; then
```

Comment:
Isn't it nicer with path filters? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore